main.go

```diff
@@ -99,3 +115,50 @@ func goToolDryRunCmd(args []string) string {
 	}
 	return b.String()
 }
+
+// isGovvvDirective returns true if the arg is a govvv directive, and false
+// otherwise.
+func isGovvvDirective(arg string) bool {
+	_, ok := govvvDirectives[arg]
+	return ok
+}
+
+// scrubGovvvDirectives filters out govvv directs to return a clean set of args
+// that can be passed to the go command.
+func scrubGovvvDirectives(args []string) (filtered []string) {
+	filtered = []string{}
+	skipping := 0
+	for _, arg := range args {
+		if skipping > 0 {
+			skipping--
+			continue
+		}
+		if hasArgument, ok := govvvDirectives[arg]; ok {
+			if hasArgument {
+				skipping = 1
+			}
+		} else {
+			filtered = append(filtered, arg)
+		}
+	}
+	return filtered
+}
+
+// collectGovvvDirective searches the args array for a directive and a possible
+// argument for that directive.  It returns that argument, or "", if the
+// directive takes none, and an indication if the directive was actually found.
+func collectGovvvDirective(args []string, directive string) (argument string, ok bool) {
```

Comment:
would be nice to have unit test for this as it's non-trivial.

Reply:
There should be 100% coverage for the new functions.